src/curve_fit.jl

```diff
@@ -1,23 +1,29 @@
 immutable LsqFitResult{T,N}
     # simple type container for now, but can be expanded later
+    model::Function
     dof::Int
     param::Vector{T}
     resid::Vector{T}
+    rsquared::Float64
+    stderror::Float64
     jacobian::Matrix{T}
     converged::Bool
     wt::Array{T,N}
 end
 
 # provide a method for those who have their own Jacobian function
-function lmfit(f::Function, g::Function, p0, wt; kwargs...)
+function lmfit(model::Function, xpts::AbstractArray, ydata::AbstractArray, f::Function, g::Function, p0, wt; kwargs...)
```

Comment:
can you explain what this is? people are supposed to pass a model *and* f and g?

Reply:
I also am confused about why you need `model` and `f`.